From stern@rowland.harvard.edu Fri Apr 14 13:44:21 2006
Date: Fri, 14 Apr 2006 16:44:11 -0400 (EDT)
From: Alan Stern <stern@rowland.harvard.edu>
To: Greg KH <greg@kroah.com>, David Brownell <david-b@pacbell.net>
Subject: USB: net2280: check for shared IRQs
Message-ID: <Pine.LNX.4.44L0.0604141642030.5320-100000@iolanthe.rowland.org>

This patch (as670) adds a check for whether a shared IRQ was actually
generated by the net2280 device.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/net2280.c |    4 ++++
 1 file changed, 4 insertions(+)

--- gregkh-2.6.orig/drivers/usb/gadget/net2280.c
+++ gregkh-2.6/drivers/usb/gadget/net2280.c
@@ -2742,6 +2742,10 @@ static irqreturn_t net2280_irq (int irq,
 {
 	struct net2280		*dev = _dev;
 
+	/* shared interrupt, not ours */
+	if (!(readl(&dev->regs->irqstat0) & (1 << INTA_ASSERTED)))
+		return IRQ_NONE;
+
 	spin_lock (&dev->lock);
 
 	/* handle disconnect, dma, and more */
